### ISSUE TITLE
Stubbed methods with default arguments not working

### ISSUE THREAD

**Issue by simaoneves** [icy-arctic-fox/spectator#43]
It seems that default arguments are not working properly when the method in question is being stubbed.

```crystal
require "spectator"

class Person
  def initialize(@dog = Dog.new)
  end

  def pet
    @dog.pet
  end
end

class Dog
  def initialize()
  end

  def pet(times = 3)
    "woof" * times
  end
end

Spectator.describe Person do
  mock Dog do
    stub pet(times)
  end

  describe "#pet" do
    it "pets the persons dog" do
      dog = Dog.new
      person = Person.new(dog)
      allow(dog).to receive(pet()).and_return("woof")

      result = person.pet

      expect(result).to be("woof")
      expect(dog).to have_received(pet())
    end
  end
end

```

The error is:
<img width="492" alt="Screenshot 2022-02-23 at 17 48 05" src="https://user-images.githubusercontent.com/1037529/155377282-5c5dcfc7-fc06-48be-8656-c824503606a5.png">

Adding another `stub` to the mock (something like `stub pet()`) also does not work.

**Comment by icy-arctic-fox** [comment]
The current spec implementation requires that the stub signature matches the original method. It effectively replaces it, and if it doesn't match, then the stub doesn't work. Clunky and not the greatest, I plan to rework the stub system.

This works for me, can you confirm?

```crystal
  mock Dog do
    stub pet(times = 3)
  end
```

**Comment by simaoneves** [comment]
> The current spec implementation requires that the stub signature matches the original method. It effectively replaces it, and if it doesn't match, then the stub doesn't work. Clunky and not the greatest, I plan to rework the stub system.
> 
> This works for me, can you confirm?
> 
> ```crystal
>   mock Dog do
>     stub pet(times = 3)
>   end
> ```

Wasn't aware that default arguments is effectively a new method signature, then it makes sense.
Your suggestions fixes the example, but only for when you want to use the value in the default argument.
Here is another example that uses a custom value and it fails:
```crystal
require "spectator"

class Person
  def initialize(@dog = Dog.new)
  end

  def pet
    @dog.pet
  end

  def pet_more
    @dog.pet(5)
  end
end

class Dog
  def initialize()
  end

  def pet(times = 2)
    "woof" * times
  end
end

Spectator.describe Person do
  mock Dog do
    stub pet(times = 2)
  end

  describe "#pet" do
    it "pets the persons dog" do
      dog = Dog.new
      person = Person.new(dog)
      allow(dog).to receive(pet()).and_return("woof")

      result = person.pet

      expect(dog).to have_received(pet()).with(2)
    end
  end

  describe "#pet_more" do
    it "pets the persons dog alot" do
      dog = Dog.new
      person = Person.new(dog)
      allow(dog).to receive(pet()).and_return("woof")

      result = person.pet_more

      expect(dog).to have_received(pet()).with(5)
    end
  end
end
```

Thanks for looking into this 😄

**Comment by icy-arctic-fox** [comment]
This issue should be resolved in v0.11.0. There is a [spec](https://github.com/icy-arctic-fox/spectator/commit/da3fbc960765aae361214e44f02fc53d6107339b) dedicated to it. Please reopen if your issue is not resolved.
